### PR TITLE
Log warning if generated class already exists

### PR DESCRIPTION
**Pull request by jaceko**

### CODE REVIEW

File: generators/generators-commons/src/test/java/uk/gov/justice/services/generators/commons/client/AbstractClientGeneratorTest.java [hunk 1]
```diff
@@ -259,7 +273,51 @@ public void shouldThrowExceptionIfActionOtherThanPOSTorGET() {
...
+                messagingRamlWithDefaults()
+                        .with(resource()
+                                .with(httpAction(GET)))
+                        .build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder,
+                        generatorProperties().withServiceComponentOf("COMMAND_CONTROLLER").build(), singletonList( existingFilePath())));
+
+        logger.removeAppender(appender);
+        final List<LoggingEvent> logEntries = appender.messages();
+        Assert.assertThat(logEntries, not(empty()));
```
Comment (mapingo) [inline generators/generators-commons/src/test/java/uk/gov/justice/services/generators/commons/client/AbstractClientGeneratorTest.java]: Static import of Assert

File: generators/generators-commons/src/main/java/uk/gov/justice/services/generators/commons/helper/GeneratedClassWriter.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                        .writeTo(configuration.getOutputDirectory());
+            } else {
+                logger.warn("The class {} already exists, skipping code generation.", typeSpec.name);
+            }
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private static boolean classDoesNotExist(final GeneratorConfig configuration, final TypeSpec typeSpec) {
```
Comment (matt-rich) [inline generators/generators-commons/src/main/java/uk/gov/justice/services/generators/commons/helper/GeneratedClassWriter.java]: Would prefer this naming to not be in the negative, so classExists as the method name, but maybe just a personal thing...

Comment (jaceko) [inline generators/generators-commons/src/main/java/uk/gov/justice/services/generators/commons/helper/GeneratedClassWriter.java]: done

File: generators/generators-commons/src/test/java/uk/gov/justice/services/generators/commons/client/AbstractClientGeneratorTest.java (hunk 1 above)
Comment (jaceko) [inline generators/generators-commons/src/test/java/uk/gov/justice/services/generators/commons/client/AbstractClientGeneratorTest.java]: done

**Pull request merged**